### PR TITLE
Revert restrictive validation behavior for DM/WM defaults in the X module

### PR DESCRIPTION
**Pull request by Ma27**
The original idea behind this change (described in ticket #11064) was to
improve the assertions to avoid that users of the X server accidentally
forget to configure a DM or WM.

However this caused several issues with setups that require X, but no DM
or WM. The keymap testcases became instable as well as now disabling DMs
needs to be done explicitly.
(see https://github.com/NixOS/nixpkgs/pull/31268#issuecomment-347080036)

In the end the idea behind the change and #11064 was obviously a
mistake, so reverting it completely for now should be fine.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Ma27** [comment]
/cc @nbp @NeQuissimus @bendlas @PierreR I did a full revert, could you please have a look at this?

**Note for the merger:** as this is subjected to fix breackage, can we **please** do a backport to `release-18.03`?

File: nixos/doc/manual/release-notes/rl-1803.xml [hunk 1]
```diff
@@ -247,14 +247,6 @@ following incompatible changes:</para>
       overhead of a go package having non-versioned dependencies.
     </para>
   </listitem>
-  <listitem>
-    <para>
-      When using <option>services.xserver.libinput</option> (enabled by default in GNOME),
```
Comment (jtojnar) [inline nixos/doc/manual/release-notes/rl-1803.xml]: This is from https://github.com/NixOS/nixpkgs/pull/31995, it should not be removed.

**Review by jtojnar — CHANGES_REQUESTED** [review]

**Comment by Ma27** [comment]
@jtojnar @bendlas @PierreR I could use some feedback as I guess that we want to have this in 18.03 %)

**Comment by nbp** [comment]
Why couldn't we use the `services.xserver.plainX` option in this case?

**Comment by bendlas** [comment]
@Ma27 I have applied this PR and everything seems fine. When looking through the patch, I didn't see anything, that was off and you already took care of @jtojnar 's feedback, right? So .. LGTM

On a personal note: I suppose it can't have been easy pulling back on this, when you already put your time into developing it. I want to thank you and commend you for being able to stay cool about everything, even when the discussion got heated. I'm glad we share a community.

@nbp my most concise answer is: Nobody seems to want to own this feature and deal with the tickets falling out from it, including applying all the necessary `plainX` across nixos. Do you?

My slightly longer answer is: You want to require users, whose configurations have worked so far, to add a flag to express that they want to keep their configuration working? From the concepts I recognize, this is called a deprecation. So you want to deprecate undeclared X sessions. For the gain of preventing newbie mistakes. OK. I have a couple of questions:

- What alternatives for preventing this particular mistake have been discussed?
- How big is this problem, in the space of common newbie mistakes?
- Why start with a hard error?
- What's the set of modules affected by this change?

@vcunat it seems, you were a proponent of https://github.com/NixOS/nixpkgs/issues/11064
Have you been following?

**Comment by vcunat** [comment]
I haven't been following this lately.  I was (and am) a "proponent" of the high-level intention &ndash; to avoid users not having any WM or DM unless they really know they don't want it.  There surely are multiple approaches how to reduce the risk, e.g. good comments in the generated initial config can cover part of that.

**Comment by Ma27** [comment]
> I have applied this PR to my system and everything seems fine. When looking through the patch, I didn't see anything, that was off and you already took care of @jtojnar 's feedback, right? So .. LGTM

Yes, I just screwed up locally due to some merge conflicts and accidentally removed the wrong entry from the release notes, it's fixed now :-)

> On a personal note: I suppose it can't have been easy pulling back on this, when you already put your time into developing it. I want to thank you and commend you for being able to stay cool about everything, even when the discussion got heated. I'm glad we share a community.

first of all I really appreciate that you add this sort of personal note! The thing is, I obviously underestimated the consequences of my patch(es) for the X server and I'd like to avoid any unnecessary breackage (several people asked to fix/revert this in the second PR so I decided to go ahead for now). In case I have more time we could discuss even more about this, but right now I unfortunately have to assign this job to other contributors.

I think that the aspect with the unnecessary high breackage @bendlas stated here and in the original PRs is why we should do the revert for now (and backport the revert to `18.03` to avoid any more unexpected breackage). Then we could reopen #11064 and continue discussing there to find a better solution for 18.09, right?

**Pull request closed without merging**